src/settingsui.ts

```diff
@@ -579,7 +599,7 @@ export function renderSettings() {
 		const customUserScriptSettingsElems: DocumentFragment[] = []
 		const userscriptSettings: RenderReadySetting[] = su.userscripts
 			.map(userscript => {
-				const obj: RenderReadySetting = {
+				const customUserscriptSetting: RenderReadySetting = {

```

Comment:
im not a huge fan of how big this is. you think you could refactor atleast some parts from the deeply nested conditionals out of there?

Reply:
I can definitely refactor to make a little more readable, but I'm not sure I could remove nested statements without a lot of code redundancy. 

Maybe I could split userscriptPreferences into a separate loop through the userscripts, but that feels a little unnecessary. I've been staring at this code for an hour and I'm having trouble figuring out how to refactor to remove nested statements without redundancy. What do you think?